GeneralsMD/Code/GameEngine/Source/GameLogic/Object/Body/UndeadBody.cpp

```diff
@@ -87,6 +87,14 @@ void UndeadBody::attemptDamage( DamageInfo *damageInfo )
 	{
 		damageInfo->in.m_amount = min( damageInfo->in.m_amount, getHealth() - 1 );
 		shouldStartSecondLife = TRUE;
+
+#if !RETAIL_COMPATIBLE_CRC
+		if (isSubdued())
+		{
+			const UndeadBodyModuleData* data = getUndeadBodyModuleData();
+			internalAddSubdualDamage(data->m_secondLifeMaxHealth - getCurrentSubdualDamageAmount());
```

Comment:
> As a Battle Bus gains 250 health when entering bunkered mode (400 health → 650 health), the subdual damage threshold can be shifted above the current subdual damage and thus cause the Battle Bus to remain subdued forever. To rectify this, the subdual damage is set to the threshold (max health) on transitioning to the bunkered state to ensure the subdued state is removed when healed.

This observation makes me wonder if it is desireable that a unit can get stuck in subdual to begin with. Where does this happen and is there maybe another way to fix this universally, so that even in these scenarios the sub dual damage can recover on its own?

Reply:
Do you know why the subdue gets stuck when the health is wrong?